hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CreatePipelineCommandHandler.java

```diff
@@ -163,6 +176,21 @@ public long getAverageRunTime() {
 
   @Override
   public int getQueuedCount() {
-    return 0;
+    return queuedCount.get();
+  }
+
+  @Override
+  public void stop() {
+    if (executor != null) {
+      try {
+        executor.shutdown();
+        if (!executor.awaitTermination(3, TimeUnit.SECONDS)) {
+          executor.shutdownNow();
+        }
+      } catch (InterruptedException ie) {
+        // Ignore, we don't really care about the failure.
+        Thread.currentThread().interrupt();
+      }
+    }
   }
```

Comment:
The executor is also shutdown in `DatanodeStateMachine`, so I think we can omit this.

```suggestion
```